end_to_end/end_to_end_suite_test.go

```diff
@@ -2652,4 +2652,197 @@ LANGUAGE plpgsql NO SQL;`)
 			Expect(string(output)).To(ContainSubstring("Multiple timestamp directories found"))
 		})
 	})
+	Describe("Backup partition these parent is in extension", func() {
+		BeforeEach(func() {
+			_ = os.Chdir("resources")
+			command := exec.Command("make", "USE_PGXS=1", "install")
+			mustRunCommand(command)
+			_ = os.Chdir("..")
+
+			testhelper.AssertQueryRuns(backupConn, `CREATE EXTENSION test_ext6;`)
+		})
+		AfterEach(func() {
+			testhelper.AssertQueryRuns(backupConn, `DROP EXTENSION test_ext6 CASCADE;`)
+		})
+
+		It("backup partition these parent is in extension", func() {
+			if backupConn.Version.AtLeast("7") {
+				testhelper.AssertQueryRuns(backupConn,
+					`
+							CREATE TABLE test_part PARTITION OF t_part FOR VALUES FROM (10) TO (20) PARTITION BY LIST(c);
+							CREATE TABLE part_a PARTITION OF test_part FOR VALUES IN ('a');
+							CREATE TABLE part_b PARTITION OF test_part FOR VALUES IN ('b');
+							CREATE TABLE part_c PARTITION OF t_part_1_prt_1 FOR VALUES IN ('c');
+
+							INSERT INTO part_a SELECT a, 15, a, 'a' FROM generate_series(1, 100)a;
+							INSERT INTO part_b SELECT a, 16, a, 'b' FROM generate_series(1, 100)a;
+							INSERT INTO part_c SELECT a, 0, a, 'c' FROM generate_series(1, 100)a;
+					`)
+			} else {
+				testhelper.AssertQueryRuns(backupConn,
+					`
+							ALTER TABLE t_part ADD PARTITION test_part START (10) END (20);
+							ALTER TABLE t_part ALTER PARTITION FOR (RANK(1)) ADD PARTITION part_c VALUES ('c');
+
+							INSERT INTO t_part_1_prt_1_2_prt_part_c SELECT a, 0, a, 'c' FROM generate_series(1, 100)a;
+							INSERT INTO t_part_1_prt_test_part_2_prt_a_part SELECT a, 15, a, 'a' FROM generate_series(1, 100)a;
+							INSERT INTO t_part_1_prt_test_part_2_prt_b_part SELECT a, 16, a, 'b' FROM generate_series(1, 100)a;
+					`)
+			}
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir, "--leaf-partition-data")
+			metadataFileContents := string(getMetdataFileContents(backupDir, timestamp, "metadata.sql"))
+
+			if backupConn.Version.AtLeast("7") {
+				Expect(metadataFileContents).To(ContainSubstring("t_part"))
+				Expect(metadataFileContents).To(ContainSubstring("part_a"))
+				Expect(metadataFileContents).To(ContainSubstring("part_b"))
+				Expect(metadataFileContents).To(ContainSubstring("part_c"))
+			} else {
+				// In 6X we will create test_part, part_a and part_b in a single DDL
+				Expect(metadataFileContents).To(ContainSubstring("test_part"))
+				Expect(metadataFileContents).To(ContainSubstring("part_c"))
+			}
+
+			gprestore(gprestorePath, restoreHelperPath, timestamp, "--backup-dir", backupDir, "--redirect-db", "restoredb")
+
+			testQuery := `select count(*) from t_part;`
```

Comment:
Test framework has some helpers to check integrity, such as assertDataRestored() and assertTablesRestored().
I think, by using them, the code will be cleaner and more, we can omit a block with ContainSubstring() above. As out patch is complex, I suggest to check all available tables here.